### PR TITLE
Use sanitize-filename to sanitize screenshot filenames

### LINKED ISSUE #3052: Filename limitation in cy.screenshot
### Description:
I was wondering if Cypress could allow CJK's characters in a filename when we call cy.screenshot() or cy.screenshot(filename). 

As the definition of source code [screenshots.coffee](https://github.com/cypress-io/cypress/blob/master/packages/server/lib/screenshots.coffee) (if this is the right place?), the valid characters in a filename  seems like
```
invalidCharsRe = /[^0-9a-zA-Z-_\s\(\)]/g
```
Besides the valid characters, all the other characters in a filename will replace to "".

```
replaceInvalidChars = (str) ->
  str.replace(invalidCharsRe, "")
```
We wrote Chinese description for each test run, for we can generate Chinese report. This cause problems when we use default naming convention  ```cy.screenshot()```, or even worse,  ```cy.screenshot(Chinese_Filename)```, as described below.

### Current behavior:

```js
// spec name = test.spec.js

context('語言', () => { // language
    it('預設', function() { // default naming rule
        cy.screenshot();
        // screenshots/test.spec.js/ -- .png
        // Will create a image named " -- .png" under the folder "test.spec.js"
    });
    it('Give shot a name', function() { // Take screen by giving a Chinese filename
        cy.screenshot('你好');
        // screenshots/test.spec.js.png
        // Will create a image named "test.spec.js.png" and
        // the created image is NOT under the folder "test.spec.js"
    });
});

```

### Desired behavior:

```js
context('語言', () => {
    it('預設', function() {
        cy.screenshot(); // screenshots/test.spec.js/語言 -- 預設.png
    });
    it('Give shot a name', function() {
        cy.screenshot('你好'); // screenshots/test.spec.js/你好.png
    });
});
```

### Something I have tried
Listen the event ```after:screenshot```, and then, I can move the created image to the correct place with correct filename. ```Cypress.Screenshot.defaults({onAfterScreenshot ($el, props) {},})``` can do the same job too. However, it will lose the information corresponding to test run's description in the above manner. This is not working if I take screenshot via default naming rule```cy.screenshot()```. I think this is just a workaround.

Is it possible to only discard the special characters in a filename without losing CJK's characters? Though, this would become not fully portable filenames, but it become more convenience for non-english users.

### Versions

Cypress 3.1.2
MacOS 10.13.4
Chrome 71.0.3578.98

### PR DESCRIPTION
**Pull request by lilaconlee**
- Fixes #3052 

From the issue
> There were some changes in kinda unrelated tests as $, ^, and % are now allowed in filenames. According to the package, this should be okay but lemme know if these were excluded for a reason.

### CODE REVIEW

File: packages/server/test/unit/screenshots_spec.coffee [hunk 1]
```diff
@@ -470,25 +470,37 @@ describe "lib/screenshots", ->
...
+        titles: ["bar", "baz"]
+        takenPaths: ["a"]
+        testFailure: true
+      }, "png", "path/to/screenshots")
+      .then (p) ->
+        expect(p).to.eq(
+          "path/to/screenshots/examples/user/list.js/bar -- baz (failed).png"
+        )
+
+    it "sanitizes file paths", ->
```
Comment (chrisbreiding) [inline packages/server/test/unit/screenshots_spec.coffee]: Part of the purpose of this test is to prove that characters that aren't allowed get sanitized. Can we put something in that's not supposed to be in a filename and ensure it gets removed?

Comment (chrisbreiding) [reply packages/server/test/unit/screenshots_spec.coffee]: Just to be clear, I'm talking about the `sanitizes file paths` spec. (GitHub highlighted the lines above the test for some reason)

Comment (lilaconlee) [reply packages/server/test/unit/screenshots_spec.coffee]: The `*` and `...` chars are being removed from the filename, are there certain characters that'd be better to test here?

Comment (chrisbreiding) [reply packages/server/test/unit/screenshots_spec.coffee]: Sorry, my bad. I misread the diff 😓

**Pull request merged**